GPU/Directx9/VertexShaderGeneratorDX9.cpp

```diff
@@ -232,7 +232,7 @@ void GenerateVertexShaderDX9(int prim, char *buffer, bool useHWTransform) {
 			WRITE(p, "%s", boneWeightAttrDecl[TranslateNumBones(vertTypeGetNumBoneWeights(vertType))]);
 		}
 		if (doTexture && hasTexcoord) {
-			if (doTextureProjection)
+			if (doTextureProjection && !throughmode)
```

Comment:
So, quick comparison of the attribute (vertex data) incoming to the vertex shader:

GLES:
vec3 iff: doTexture && hasTexcoord && !useHWTransform && doTextureProjection
vec3 otherwise

DX9:
float3 if: !useHWTransform
float3 if: doTexture && hasTexcoord && useHWTransform && doTextureProjection
float2 otherwise

This logic seems very different to me.

-[Unknown]